packages/template-ui/src/components/ContentImage.vue

```diff
@@ -31,6 +38,36 @@ export default {
       required: true,
     },
   },
+  data() {
+    return {
+      imageSize: 0,
+    };
+  },
+  computed: {
+    ...mapState(['bundleKind']),
+    buttonStyle() {
+      const min = contentImageMinWidth;
+      const maxWidth = this.imageSize >= min ? this.imageSize : min;
+      return {
+        maxWidth,
+      };
+    },
+    cardStyle() {
+      const margin = cardSpacerX;
+      return {
+        maxWidth: `calc(${this.imageSize}px + ${margin})`,
```

Comment:
Yeah, from the CSS is not possible to do things conditionally for the image size. This is not a good practice but is what design wants today.